JASP-Engine/rbridge.cpp

```diff
@@ -745,13 +744,43 @@ std::string	rbridge_encodeColumnNamesToBase64(const std::string & filterCode)
 				if(filterColumnsUsed.count(col) == 0)
 					filterColumnsUsed.insert(col);
 			}
-		}
-	}
+        }
+    }
 
-	//Log::log() << " rbridge_encodeColumnNamesToBase64 results in: "<<filterBase64 << std::endl;
+    //Log::log() << " rbridge_encodeColumnNamesToBase64 results in: "<<filterBase64 << std::endl;
 
-	return filterBase64;
+    return filterBase64;
 }
+std::vector<int> rbridge_getPositionsColumnNameMatches(const std::string & filterBase64, const std::string & columnName)
+{
+    std::vector<int> positions;
+    bool inString = false;
+    char delim;
+    for (std::string::size_type pos = 0; pos < filterBase64.length(); ++pos)
+    {
+		if (!inString && filterBase64.substr(pos, columnName.length()) == columnName)
+		{
+        	positions.push_back(int(pos));
+			continue;
+		}
```

Comment:
Can we not get rid of the continue and just do an `} else if (filterBase64[pos] etc`?

Like:
```
for (std::string::size_type pos = 0; pos < filterBase64.length(); ++pos)
	if (!inString && filterBase64.substr(pos, columnName.length()) == columnName)
		positions.push_back(int(pos));
	else if (filterBase64[pos] == '"' || filterBase64[pos] == '\'')
	{
		if (inString && filterBase64[pos] == delim)
			inString = false;
		else if (!inString)
		{
			delim = filterBase64[pos];
			inString = true;
		}
	}

return positions;
}
```

Reply:
I think this is more readable an clearly shows that you cannot get to the second if-block if you add a position.